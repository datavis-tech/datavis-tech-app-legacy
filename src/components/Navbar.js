import React from 'react'
import { Link, IndexLink } from 'react-router'
import { connect } from 'react-redux'

const Navbar = ({ isLoggedIn }) => (
  <nav className="navbar navbar-light bg-faded">

    <div className="container">

      <IndexLink to="/" className="navbar-brand">
        Datavis.tech
      </IndexLink>

      <ul className="nav navbar-nav">

        <li className="nav-item">
          <IndexLink to="/" className="nav-link" activeClassName="active">
            Home
          </IndexLink>
        </li>

        <li className="nav-item">
          <Link to="/consulting" className="nav-link" activeClassName="active">
            Consulting
          </Link>
        </li>

        <li className="nav-item">
          <Link to="/about" className="nav-link" activeClassName="active">
            About
          </Link>
        </li>

      </ul>
      <form className="form-inline float-xs-right">
        <Link to="/create" className="btn btn-primary" activeStyle={{display: 'none'}}>
          Create
        </Link> {
          isLoggedIn ? (
            <a href="/api/auth/logout" className="btn btn-primary">Log out</a>
          ) : (
            <a href="/api/auth/github" className="btn btn-primary">Log in</a>
          )
        }
      </form>
    </div>
  </nav>
)

const mapStateToProps = ({ user: {isLoggedIn}}) => ({ isLoggedIn })

export default connect(mapStateToProps)(Navbar)
