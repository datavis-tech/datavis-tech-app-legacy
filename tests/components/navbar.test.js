import React from 'react'
import { shallow } from 'enzyme'
import fakeUser from '../utils/fakeUser'

import Navbar from '../../src/components/spacer'

describe('spacer', () => {

  it(`should match snapshot without user`, () => {
    const sut = shallow(<Navbar />)
    expect(sut).toMatchSnapshot()
  })

  it(`should match snapshot with user`, () => {
    const sut = shallow(<Navbar user={fakeUser}/>)
    expect(sut).toMatchSnapshot()
  })

})
