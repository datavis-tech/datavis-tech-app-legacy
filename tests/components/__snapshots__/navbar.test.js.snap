// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar should match snapshot with user 1`] = `
<Menu
  secondary={true}
>
  <MenuItem>
    <LinkRoutes
      route="/"
    >
      <a>
        <Image
          as="img"
          height="36px"
          src="/static/images/Logo_Nav.png"
          ui={true}
        />
      </a>
    </LinkRoutes>
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <CreateMenu
      user={
        Object {
          "_json": Object {
            "avatar_url": "https://avatars2.githubusercontent.com/u/156229?v=4",
          },
          "displayName": "Fred",
          "id": undefined,
          "subscriptionPlan": undefined,
          "username": "fred",
        }
      }
    />
    <mockConstructor
      user={
        Object {
          "_json": Object {
            "avatar_url": "https://avatars2.githubusercontent.com/u/156229?v=4",
          },
          "displayName": "Fred",
          "id": undefined,
          "subscriptionPlan": undefined,
          "username": "fred",
        }
      }
    />
  </MenuMenu>
</Menu>
`;

exports[`navbar should match snapshot without user 1`] = `
<Menu
  secondary={true}
>
  <MenuItem>
    <LinkRoutes
      route="/"
    >
      <a>
        <Image
          as="img"
          height="36px"
          src="/static/images/Logo_Nav.png"
          ui={true}
        />
      </a>
    </LinkRoutes>
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <CreateMenu />
    <mockConstructor />
  </MenuMenu>
</Menu>
`;
